src/test/java/net/snowflake/client/config/SFConnectionConfigParserTest.java

```diff
@@ -103,6 +105,21 @@ public void testThrowErrorWhenWrongPermissionsForTokenFile() throws IOException
         SnowflakeSQLException.class, () -> SFConnectionConfigParser.buildConnectionParameters());
   }
 
+  @Test
+  public void testNoThrowErrorWhenWrongPermissionsForTokenFileButSkippingFlagIsEnabled()
+      throws SnowflakeSQLException, IOException {
+    SnowflakeUtil.systemSetEnv(SNOWFLAKE_HOME_KEY, tempPath.toString());
```

Comment:
Should we reset these envs at the end?

Reply:
Do you mean something more like  SnowflakeUtil.systemUnsetEnv (done in @After method)?

Reply:
Or maybe not unset, rather reset to the value from before changing?

Reply:
Added code to remember env values and reset them at the end.